client/environment.go

```diff
@@ -138,6 +138,7 @@ type EnvironmentCreate struct {
 	IsRemoteBackend             *bool                  `json:"isRemoteBackend,omitempty" tfschema:"-"`
 	Type                        string                 `json:"type,omitempty"`
 	DriftDetectionRequest       *DriftDetectionRequest `json:"driftDetectionRequest,omitempty" tfschema:"-"`
+	PreventAutoDeploy           *bool                  `json:"preventAutoDeploy,omitempty" tfschema:"-"`
```

Comment:
why is it a pointer, is it okay to always pass false?

Reply:
if it won't be a pointer and the user won't specify it - will it evaluate to false? I think we could always pass it but is it really better?

Reply:
it simplifies the code significantly.

But it depends on the API. Some of the API's params do have "3" values: "pass nothing", "false", "true".

Reply:
I can try passing false as default. let me see

Reply:
thanks!

Reply:
I changed it anyway cause it should be simpler - can you take a quick look?

Reply:
unit test is failing...

Reply:
yea I gave up on the `bool` change, it failing to much tests now 😅